docs/markdown/FAQ.md

```diff
@@ -500,3 +500,78 @@ meson -Dcpp_eh=none -Dcpp_rtti=false <other options>
 ```
 
 The RTTI option is only available since Meson version 0.53.0.
+## Why doesn't meson have user defined functions/macros? I **need** them!
+
+The tl;dr answer to this is that meson's design is focused on solving specific
+problems rather than providing a toolbox for developers to cobble something
+together themselves. Build systems should be quick to write and quick to
+understand, functions muddle this simplicity.
+
+The long answer is twofold:
+
+First, Meson wants to provide a rich set of tools that solve specific problems
+simply out of the box. This is similar to the "batteries included" mentality of
+Python. By providing tools that solve common problems in the simplest way
+possible *in* Meson we are solving that problem for everyone instead of forcing
+everyone to solve that problem for themselves over and over again, often
+badly. One example of this are Meson's dependency wrappers around various
```

Comment:
I don't think it's a problem to point out that given billions of human beings doing something, often it will be done badly if there's no authoritative "right" way to do it.

It's also an important perspective on why meson wants to unify central handling of this.

Reply:
Right now this FAQ entry read why too much opinionated, IMHO. It is never a good idea to explain that a solution is good on the base that other solutions are bad. I was trying to suggest better wording, but it has been merged already and the comments I didn't submit got lost.

Reply:
It has not been merged yet...

Reply:
I've taken the rest of your suggestions, but I'm going to leave this. The fact that it will be implemented badly by some subset of the people implementing it is an important reason to have the code in Meson and not implemented in the DSL in your project.